File: frontend/src/types/exports.ts
export type DatabaseExportJobStatus = 'queued' | 'running' | 'completed' | 'failed';

export interface DatabaseExportJobRead {
  created_at: string;
  download_url: string | null;
  error_message: string | null;
  finished_at: string | null;
  id: string;
  manifest: Record<string, unknown>;
  progress: number;
  requested_by_id: string;
  row_count: number;
  started_at: string | null;
  status: DatabaseExportJobStatus;
  table_count: number;
  updated_at: string;
}

export const DATABASE_EXPORT_STATUS_LABELS: Record<DatabaseExportJobStatus, string> = {
  completed: '已完成',
  failed: '失败',
  queued: '排队中',
  running: '执行中',
};

File: backend/app/schemas/exports.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.models.exports import DatabaseExportJobStatus


class DatabaseExportJobRead(BaseModel):
    id: UUID
    requested_by_id: UUID
    status: DatabaseExportJobStatus
    progress: int
    table_count: int
    row_count: int
    download_url: str | None
    manifest: dict[str, object]
    error_message: str | None
    created_at: datetime
    updated_at: datetime
    started_at: datetime | None
    finished_at: datetime | None

    model_config = ConfigDict(from_attributes=True)
